src/libs/actions/Report.js

```diff
@@ -616,6 +617,29 @@ function createOptimisticChatReport(participantList) {
     };
 }
 
+/*
+ * Creates an optimistic IOU report with a randomly generated reportID
+ */
+function buildOptimisticIOUReport(total, chatReportID, currency, locale) {
```

Comment:
So this function won't look for an existing open IOU?

Reply:
I think it should, but we might as well not use this function for it when we already have existing IOU report. We can just take it from onyx and put it into the optimistic data.

I dont think we should use this function to create existing optimistic objects (they already have the correct shape in Onyx)

Reply:
Correct, that logic is yet to be added. THis simply generates the correctly formatted optimistic object